src/components/tasks/Monday.tsx

```diff
@@ -0,0 +1,327 @@
+  highlightedText: string;
+}
+
+const QUERY_MONDAY_WORKSPACES = graphql(/* GraphQL */ `
+  query MondayWorkspaces($userModuleId: ID!) {
+    mondayWorkspaces(userModuleId: $userModuleId) {
+      elements {
+        id
+        name
+      }
+    }
+  }
+`);
+
+const QUERY_MONDAY_RESOURCES = graphql(/* GraphQL */ `
+  query MondayResources($userModuleId: ID!, $workspaceId: ID!) {
+    mondayResources(userModuleId: $userModuleId, workspaceId: $workspaceId) {
+      boards {
+        elements {
+          id
+          name
+          groups {
+            elements {
+              id
+              name
+            }
+          }
+        }
+      }
+      folders {
+        elements {
+          id
+          name
+          color
+          boards {
+            elements {
+              id
+              name
+              groups {
+                elements {
+                  id
+                  name
+                }
+              }
+            }
+          }
+          folders {
+            elements {
+              id
+              name
+              color
+              boards {
+                elements {
+                  id
+                  name
+                  groups {
+                    elements {
+                      id
+                      name
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+const Monday = (props: MondayProps) => {
+  const { userModuleId, highlightedText } = props;
+  const { register, setValue, control } =
+    useFormContext<MutationCreateMondayItemArgs>();
+
+  const [selectedWorkspace, setSelectedWorkspace] = useState("");
+  const [selectedGroup, setSelectedGroup] = useState<Group>({
+    id: "",
+    name: "",
+  });
+  const [openAutocomplete, setOpenAutocomplete] = useState(false);
+  const { data: mondayWorkspacesData } = useQuery(QUERY_MONDAY_WORKSPACES, {
+    variables: {
+      userModuleId,
+    },
+  });
+  const [
+    fetchMondayResources,
+    { data: mondayResourcesData, loading: mondayResourcesLoading },
+  ] = useLazyQuery(QUERY_MONDAY_RESOURCES);
+  register("userModuleId", { value: userModuleId });
+
+  useEffect(() => {
+    if (selectedWorkspace) {
+      setSelectedGroup({ id: "", name: "" });
+      fetchMondayResources({
+        variables: { userModuleId, workspaceId: selectedWorkspace },
+      });
+    }
+  }, [selectedWorkspace, fetchMondayResources, userModuleId]);
+
+  const generateGroupTree = (groups: PaginatedGroups, board: Board) => {
+    return groups.elements?.map((group) => (
+      <StyledTreeItem
+        key={`${board.id}-${group.id}`}
+        nodeId={`${board.id}-${group.id}`}
+        labelText={group.name}
+        labelIcon={FormatListBulletedOutlined}
+        onClick={() => {
+          setValue("element.boardId", board.id);
+          setValue("element.groupId", group.id);
+          setSelectedGroup({ ...group, board });
```

Comment:
Instead of using `setValue`, could we use a `Controller` with a `watch` top update the values?

Reply:
I see what you mean, but isn't too heavy because of tree for the FE ?